src/full/encoder.rs

```diff
@@ -137,6 +170,12 @@ impl Encoder {
     ///
     /// Returns `RLNCError::CodingVectorLengthMismatch` if the length of the
     /// provided `coding_vector` does not match `self.piece_count`.
+    ///
+    /// # Deprecated
+    /// Use the allocation friendly `code_with_buf_coding_vector` instead.
+    /// This method is deprecated due to performance and memory usage concerns.
+    /// It uses a vector allocation for each coded piece, which can be inefficient.
+    #[deprecated(since = "0.9.0", note = "Use `code_with_buf_coding_vector` instead for better performance and memory usage.")]
```

Comment:
This deprecation notice is helpful. We are missing one for `code_with_coding_vector` function at line 115, which is enabled when `feature != "parallel"`.